test/clients/azure/client.go

```diff
@@ -18,19 +14,11 @@ import (
 	"github.com/openshift/openshift-azure/pkg/util/configblob"
 )
 
-// rpFocus represents the supported RP APIs which e2e tests use to create their azure clients,
-// The client will be configured to work either against the real, fake or admin apis
-type rpFocus string
-
-var (
-	adminRpFocus = rpFocus(regexp.QuoteMeta("[Admin]"))
-	fakeRpFocus  = rpFocus(regexp.QuoteMeta("[Fake]"))
-	realRpFocus  = rpFocus(regexp.QuoteMeta("[Real]"))
-)
+// FakeRPClient is a singleton Azure Client used by all tests which run against the FakeRP
+var FakeRPClient *Client
 
-func (tf rpFocus) match(focusString string) bool {
-	return strings.Contains(focusString, string(tf))
-}
+// RealRPClient is a singleton Azure Client used by all tests which run against the RealRP
+var RealRPClient *Client
```

Comment:
sorry, this is almost certainly what we agreed on, but it still looks clumsy.  Given that we don't have any tests right now that can run against both fake and real RPs, how about making one top level `var SomeName *Client` having three fields within it - something like RealOpenShiftManagedClusters, FakeOpenShiftManagedClusters, FakeOpenShiftManagedClustersAdmin ?